### PR TITLE
[20414] Add PR template 

### PR DESCRIPTION
**Pull request by JesusPoderoso**
## Description

This PR adds this PR template

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- **N/A** Any new/modified methods have been properly documented using Doxygen. 
- **N/A** Tests pass locally.
- **N/A** Applicable backports have been included in the description.

## Reviewer Checklist
- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- [x] Check CI results: changes do not issue any warning.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+<!-- Fixes #(issue) -->
+
+<!-- In case the changes are built over a previous pull request, please uncomment next line. -->
+<!-- This PR depends on #(PR) and must be merged after that one. -->
+
+## Contributor Checklist
+- [ ] Commit messages follow the project guidelines. <!-- External contributors should sign the DCO. Fast DDS developers must also refer to the internal Redmine task. -->
+- [ ] Any new/modified methods have been properly documented using Doxygen. <!-- Even internal classes, and private methods and members should be documented, not only the public API. -->
+- [ ] Tests pass locally.
+- [ ] Applicable backports have been included in the description.
```
Comment (EduPonz) [inline .github/pull_request_template.md:32]: I think API and ABI compatibility and also quite important here. I'd also include the cross emoji that we have on Fast DDS. In fact, this list could be almost the same as the one in Fast DDS

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+
+## Reviewer Checklist
+- [ ] The PR has a milestone assigned.
```
Comment (EduPonz) [inline .github/pull_request_template.md:36]:
```suggestion
## Reviewer Checklist

- [ ] The PR has a milestone assigned.
```

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+- [ ] The title and description correctly express the PR's purpose.
+- [ ] Check contributor checklist is correct.
+- [ ] Check CI results: changes do not issue any warning.
```
Comment (EduPonz) [inline .github/pull_request_template.md:39]: I think CI should pass as well

**Review by EduPonz — CHANGES_REQUESTED** [review]

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Linux/397/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Mac/296/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Windows/378/)

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -26,14 +26,25 @@
...
 
 ## Contributor Checklist
+
+<!--
+    - If any of the elements of the following checklist is not applicable, substitute the checkbox [ ] by _N/A_:
+    - If any of the elements of the following checklist is not fulfilled on purpose, please provide a reason and substitute the checkbox [ ] with ❌: or __NO__:.
+-->
+
 - [ ] Commit messages follow the project guidelines. <!-- External contributors should sign the DCO. Fast DDS developers must also refer to the internal Redmine task. -->
+- [ ] The code follows the style guidelines of this project. <!-- Please refer to the [Quality Declaration](https://github.com/eProsima/Fast-DDS/blob/master/QUALITY.md#linters-and-static-analysis-4v) for more information. -->
```
Comment (EduPonz) [inline .github/pull_request_template.md:36]: Incorrect link in commit

**Review by EduPonz — CHANGES_REQUESTED** [review]

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Linux/398/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Mac/297/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Windows/379/)

**Review by EduPonz — APPROVED** [review]

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Linux/404/)
* Mac Build Status
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Windows/386/)

**Pull request merged**